src/components/users/controllers.tsx

```diff
@@ -95,3 +106,208 @@ export async function getUser(
     ),
   };
 }
+
+export async function resetPasswordRequestToken(ctx: IContext, _params: IParameters): Promise<IResponse> {
+  const template = new Template(ctx.viewContext, 'Request password reset');
+
+  return await Promise.resolve({
+    body: template.render(<PasswordResetRequest csrf={ctx.viewContext.csrf} />),
+  });
+}
+
+export async function resetPasswordObtainToken(
+  ctx: IContext, params: IParameters, body: IPasswordResetBody,
+): Promise<IResponse> {
+  const VALID_EMAIL = /^[^@]*@[^@]*$/;
+  const email = (/* istanbul ignore next */ body.email || '').toLowerCase();
+  const template = new Template(ctx.viewContext, 'Request password reset');
+
+  if (!email || !VALID_EMAIL.test(email)) {
+    template.title = 'Error: Request password reset';
+
+    return {
+      body: template.render(<PasswordResetRequest csrf={ctx.viewContext.csrf} invalidEmail={true} />),
+      status: 400,
+    };
+  }
+
+  const uaa = new UAAClient({
+    apiEndpoint: ctx.app.uaaAPI,
+    clientCredentials: {
+      clientID: ctx.app.oauthClientID,
+      clientSecret: ctx.app.oauthClientSecret,
+    },
+  });
+
+  const accounts = new AccountsClient({
```

Comment:
Only reason that wasn't here since beginning, is that there is a potential for people using CLI not the admin UI to perform certain actions, such as user creation. Meaning our own collection, could get out of date.

I'm happy for it to stay, but it's weird we have two DBs with the same data :trollface:

Reply:
UAA doesn't always store a user's email address

PaaS accounts does

Only operators have permissions to create users without going via Pazmin

Reply:
You're right, but we as operators are also eligible to make a mistake :)

Reply:
I tripped and fell and ran `curl https://uaa.london.cloud.service.gov.uk/Users -X POST -d '{...}'` 😈

Reply:
I'm glad you find it funny :D I happen to have done that once after my rejoin xD guilty

Reply:
more precisely, `uaa create-user` :P